modules/afsmtp/afsmtp.c

```diff
@@ -119,51 +116,56 @@ afsmtp_dd_set_port(LogDriver *d, gint port)
 }
 
 void
-afsmtp_dd_set_subject(LogDriver *d, const gchar *subject)
+afsmtp_dd_set_subject(LogDriver *d, const gchar *subject, GlobalConfig *cfg)
 {
   AFSMTPDriver *self = (AFSMTPDriver *)d;
 
-  g_free(self->subject);
-  self->subject = g_strdup(subject);
+  log_template_unref(self->subject_tmpl);
+  self->subject_tmpl = log_template_new(cfg, "subject");
+  self->subject_tmpl->template = g_strdup(subject);
 }
```

Comment:
Hmmm why do you need to poke into that member?

This should probably be a log_template_compile call. Or even better use the template_content rule in the grammar instead of a string, that will give you proper error handling as well.

What happens if an invalid template is specifiied?